alacritty/src/macos/proc.rs

```diff
@@ -1,41 +1,84 @@
-use std::ffi::{CStr, CString};
+use std::ffi::{CStr, CString, IntoStringError};
+use std::fmt::{self, Display, Formatter};
 use std::io;
 use std::mem::{self, MaybeUninit};
-use std::os::raw;
+use std::os::raw::{c_int, c_void};
 use std::path::PathBuf;
 
-pub fn cwd(pid: raw::c_int) -> io::Result<PathBuf> {
+/// Error occuring during getting cwd from `proc_pidinfo`.
+#[derive(Debug)]
+pub enum Error {
+    /// IO error using `proc_pidinfo`.
+    Io(io::Error),
+
+    /// Error converting into utf8 string.
+    IntoString(IntoStringError),
+
+    /// Invalid size.
+    InvalidSize,
+}
+
+impl std::error::Error for Error {
+    fn source(&self) -> Option<&(dyn std::error::Error + 'static)> {
+        match self {
+            Error::InvalidSize => None,
+            Error::Io(err) => err.source(),
+            Error::IntoString(err) => err.source(),
+        }
+    }
+}
+
+impl Display for Error {
+    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
+        match self {
+            Error::InvalidSize => write!(f, "Invalid proc_pidinfo return size."),
+            Error::Io(err) => write!(f, "Error getting current working directory: {}", err),
+            Error::IntoString(err) => {
+                write!(f, "Error when parsing current working directory: {}", err)
+            },
+        }
+    }
+}
+
+impl From<io::Error> for Error {
+    fn from(val: io::Error) -> Self {
+        Error::Io(val)
+    }
+}
+
+impl From<IntoStringError> for Error {
+    fn from(val: IntoStringError) -> Self {
+        Error::IntoString(val)
+    }
+}
+
+pub fn cwd(pid: c_int) -> Result<PathBuf, Error> {
     let mut info = MaybeUninit::<sys::proc_vnodepathinfo>::uninit();
-    let info_ptr = info.as_mut_ptr() as *mut raw::c_void;
-    let size = mem::size_of::<sys::proc_vnodepathinfo>() as raw::c_int;
+    let info_ptr = info.as_mut_ptr() as *mut c_void;
+    let size = mem::size_of::<sys::proc_vnodepathinfo>() as c_int;
 
     let c_str = unsafe {
         let pidinfo_size = sys::proc_pidinfo(pid, sys::PROC_PIDVNODEPATHINFO, 0, info_ptr, size);
         match pidinfo_size {
-            c if c < 0 => return Err(io::Error::last_os_error()),
-            s if s != size => {
-                return Err(io::Error::new(io::ErrorKind::Other, "invalid return size"))
-            },
+            c if c < 0 => return Err(io::Error::last_os_error().into()),
+            s if s != size => return Err(Error::InvalidSize),
             _ => CStr::from_ptr(info.assume_init().pvi_cdir.vip_path.as_ptr()),
         }
     };
 
-    CString::from(c_str)
-        .into_string()
-        .map(PathBuf::from)
-        .map_err(|e| io::Error::new(io::ErrorKind::Other, e))
+    Ok(CString::from(c_str).into_string().map(PathBuf::from)?)
 }
 
-// Cleaned up `bindgen` output below
+/// Bindings for libproc `proc_pidinfo` function.
```

Comment:
```suggestion
/// Bindings for libproc.
```

While currently our goal is just the one function, they certainly provide more than just that.